### PR TITLE
props module - BaseController work

### PR DESCRIPTION
**Pull request by frankhuster**
And, funny thing, the API I'm working on is still not ready...

### CODE REVIEW

File: api/pom.xml [hunk 1]
```diff
@@ -48,6 +48,12 @@
 
         <dependency>
             <groupId>org.motechproject.nms</groupId>
+            <artifactId>props</artifactId>
```
Comment (llavoie) [inline api/pom.xml]: Based on this morning's conversation, should we change the name to config or configuration?

File: api/src/test/java/org/motechproject/nms/api/osgi/UserControllerBundleIT.java [hunk 2]
```diff
@@ -411,35 +410,36 @@ public void testSetLanguageInvalidCallingNumber() throws IOException, Interrupte
...
+        StringEntity params = new StringEntity(
+                "{\"callingNumber\":abcdef,\"callId\":\"123456789012345\",\"languageLocationCode\":\"10\"}");
         httpPost.setEntity(params);
 
         httpPost.addHeader("content-type", "application/json");
 
-        assertTrue(SimpleHttpClient.execHttpRequest(httpPost, HttpStatus.SC_BAD_REQUEST,
-                "{\"failureReason\":\"<callingNumber: Invalid>\"}",
-                ADMIN_USERNAME, ADMIN_PASSWORD));
+        assertTrue(SimpleHttpClient.execHttpRequest(httpPost, HttpStatus.SC_BAD_REQUEST, ADMIN_USERNAME,
```
Comment (llavoie) [inline api/src/test/java/org/motechproject/nms/api/osgi/UserControllerBundleIT.java]: Do we eventually want to validate the body in these tests in addition to the returned error code?

**Comment by llavoie** [comment]
@frankhuster Code looks good -- let's let @larubbio weigh in on whether props should be its own module or merged with config. 

Also, this can't be merged automatically due to conflicts.

**Comment by larubbio** [comment]
My vote would be to merge it with config.

**Pull request merged**